### PR TITLE
Deselect list item automatically

### PR DESCRIPTION
**Pull request by ghost**
when go back to systems collection view, deselect current item.

Half way from success because still can not select same item 2 times...checking why

### CODE REVIEW

File: challengeMk2/ViewModels/StarSystemsViewModel.cs [hunk 1]
```diff
@@ -126,13 +129,15 @@ List<StarSystem> GetOfflineData(string savedFile)
 
             if (File.Exists(savedFile))  // User has already saved datas when he has internet connection
```
Comment (jeremykairis) [inline challengeMk2/ViewModels/StarSystemsViewModel.cs]: Clarify the code to remove comments.

File: challengeMk2/Views/StarSystemsPage.xaml.cs [hunk 2]
```diff
@@ -31,6 +31,8 @@ protected override void OnAppearing()
 
             var vm = BindingContext as StarSystemsViewModel;
 
+            SystemCollection.SelectedItem = null;
```
Comment (jeremykairis) [inline challengeMk2/Views/StarSystemsPage.xaml.cs]: What is `//To trigger Auto PullToRefresh collection` ?

**Review by jeremykairis — APPROVED** [review]

File: challengeMk2/Views/StarSystemsPage.xaml.cs (hunk 2 above)
Comment (jeremykairis) [inline challengeMk2/Views/StarSystemsPage.xaml.cs]: By the way, selectedItem should be set to null in the setter.

**Review by jeremykairis — CHANGES_REQUESTED** [review]

File: challengeMk2/Views/StarSystemsPage.xaml.cs (hunk 2 above)
Comment (ghost) [reply challengeMk2/Views/StarSystemsPage.xaml.cs]: It is set in the setter now and allows to reselect the same item without keep the selection when go back in the list view.

File: challengeMk2/ViewModels/StarSystemsViewModel.cs [hunk 3]
```diff
@@ -30,7 +30,11 @@ public StarSystem SelectedSystem
         {
             set
             {
-                if (value != null)
+                if (value == null)
```
Comment (jeremykairis) [inline challengeMk2/ViewModels/StarSystemsViewModel.cs]: Could be improved using the early return pattern.

File: challengeMk2/Views/StarSystemsPage.xaml.cs [hunk 4]
```diff
@@ -31,7 +31,8 @@ protected override void OnAppearing()
 
             var vm = BindingContext as StarSystemsViewModel;
 
-            SystemCollection.SelectedItem = null;
+            //SystemCollection.SelectedItem = null;
+            vm.SelectedSystem = null;
```
Comment (jeremykairis) [inline challengeMk2/Views/StarSystemsPage.xaml.cs]: It should be removed from here. (Encapsulation)

**Review by jeremykairis — DISMISSED** [review]

**Pull request merged**